rasa/shared/core/trackers.py

```diff
@@ -273,16 +274,30 @@ def freeze_current_state(state: State) -> FrozenState:
             }.items()
         )
 
-    def past_states(self, domain: Domain) -> List[State]:
+    def past_states(
+        self,
+        domain: Domain,
+        ignore_rule_only_turns: bool = False,
+        rule_only_slots: Optional[List[Text]] = None,
+        rule_only_loops: Optional[List[Text]] = None,
+    ) -> List[State]:
         """Generate the past states of this tracker based on the history.
 
         Args:
-            domain: a :class:`rasa.shared.core.domain.Domain`
+            domain: a :class:`rasa.shared.core.domain.Domain`.
+            ignore_rule_only_turns: If True ignore dialogue turns that are present
+                only in rules.
+            rule_only_slots: The list of slot names,
+                which only occur in rules but not in stories.
+            rule_only_loops: The list of loop names,
+                which only occur in rules but not in stories.
```

Comment:
optional: Let's avoid encoding the types (e.g. `list`) in the docstring. The type annotations do a better job at this and don't outdate when we e.g. change this to a `Set`. To indicate a collection I'd use the plural.